erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -1599,3 +1599,31 @@ def validate_sample_quantity(item_code, sample_quantity, qty, batch_no = None):
 			format(max_retain_qty, batch_no, item_code), alert=True)
 		sample_quantity = qty_diff
 	return sample_quantity
+
+def raw_material_update_on_bom():
+	"""
```

Comment:
Please follow the docstring format. Also docstrings are not enough, you have to comment out your code that explains the blocks of code